tests/transports-integrations/tests/utils/common.py

```
@@ -392,44 +461,178 @@ def assert_valid_image_response(response: Any):
+    )
+
+    # Validate status code if available (should be 4xx for client errors)
+    if status_code is not None:
+        assert (
+            400 <= status_code < 500
+        ), f"Expected 4xx status code for invalid role error, got {status_code}"
+
+    return True
+
+
+def assert_error_propagation(error_response: Any, integration: str):
```

Ruff (0.11.9)

Missing return type annotation for public function `assert_error_propagation`

Add return type annotation: `bool`

(ANN201)